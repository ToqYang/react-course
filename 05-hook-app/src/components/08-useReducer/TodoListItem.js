import React from "react";
import "./todolistitem.css";
import { PropTypes } from "prop-types";

const TodoListItem = ({ todo = {}, index, handleDelete, handleToggle }) => {
  return (
    <li className="list-group-item">
      <p
        className={todo.done ? "complete" : ""}
        onClick={() => handleToggle(todo.id)}
      >
        {index + 1}. {todo.desc}
      </p>
      <button className="btn btn-danger" onClick={() => handleDelete(todo.id)}>
        Borrar
      </button>
    </li>
  );
};

TodoListItem.propTypes = {
  todos: PropTypes.array,
  index: PropTypes.number,
  handleDelete: PropTypes.func,
  handleToggle: PropTypes.func,
};

export default TodoListItem;
