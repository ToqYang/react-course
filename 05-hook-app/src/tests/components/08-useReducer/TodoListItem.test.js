import React from "react";
import { shallow } from "enzyme";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Test to component <TodoListItem />", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const index = 0;
  let wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      index={index}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(
      <TodoListItem
        todo={demoTodos[0]}
        index={0}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
      />
    );
  });

  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe llamar la funcion handleDelete", () => {
    wrapper.find("button").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("Debe llamar la funcion handleToggle", () => {
    wrapper.find("p").simulate("click");
    expect(handleToggle).toHaveBeenLastCalledWith(demoTodos[0].id);
  });
  test("Mostrar correctamente", () => {
    expect(wrapper.find("p").text()).toEqual(
      `${index + 1}. ${demoTodos[0].desc}`
    );
  });

  test("Complete la clase si el TODO.done = true", () => {
    const newTodo = demoTodos[0];
    newTodo.done = true;
    wrapper = shallow(<TodoListItem todo={newTodo} />);

    expect(wrapper.find("p").hasClass("complete")).toBe(true);
  });
});
