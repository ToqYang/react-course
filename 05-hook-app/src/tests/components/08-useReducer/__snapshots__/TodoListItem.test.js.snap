// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to component <TodoListItem /> Debe mostrarse correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className=""
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
